CRM/MembershipExtras/Job/UpdateMembershipPeriod.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class CRM_MembershipExtras_Job_UpdateMembershipPeriod {
```

Comment:
I don't think UpdateMembershipPeriod is a good name for such scheduled job, it is very generinc name where the job itself only process periods with overdue payments, same applies for the scheduled job name and the apiv3 class, maybe for this class and for example : CRM_MembershipExtras_Job_OverdueMembershipPeriodProcessor is a better name